src/cloud_provider/service.rs

```diff
@@ -470,6 +471,68 @@ where
     Ok(())
 }
 
+pub fn scale_down_database(
+    target: &DeploymentTarget,
+    service: &impl Database,
+    replicas_count: usize,
+) -> Result<(), EngineError> {
+    let (kubernetes, environment) = match target {
+        DeploymentTarget::ManagedServices(_, _) => {
+            info!("Doing nothing for pause database as it is a managed service");
+            return Ok(());
+        }
+        DeploymentTarget::SelfHosted(k, env) => (*k, *env),
+    };
+
+    let scaledown_ret = kubectl_exec_scale_replicas(
+        kubernetes.config_file_path()?,
+        kubernetes.cloud_provider().credentials_environment_variables(),
+        environment.namespace(),
+        Statefulset,
+        service.sanitized_name().as_str(),
+        replicas_count as u32,
+    );
+
+    cast_simple_error_to_engine_error(
+        service.engine_error_scope(),
+        service.context().execution_id(),
+        scaledown_ret,
+    )
+}
+
+pub fn scale_down_stateless_service(
+    target: &DeploymentTarget,
+    service: &impl StatelessService,
+    replicas_count: usize,
+    scaling_kind: ScalingKind,
+) -> Result<(), EngineError> {
+    let (kubernetes, environment) = match target {
+        DeploymentTarget::ManagedServices(_, _) => {
+            return Err(EngineError {
+                cause: EngineErrorCause::Internal,
+                scope: EngineErrorScope::Engine,
+                execution_id: service.context().execution_id().to_string(),
+                message: Some(format!("Cannot scale down managed service: {}", service.name_with_id())),
+            })
+        }
+        DeploymentTarget::SelfHosted(k, env) => (*k, *env),
+    };
+    let scaledown_ret = kubectl_exec_scale_replicas(
+        kubernetes.config_file_path()?,
+        kubernetes.cloud_provider().credentials_environment_variables(),
+        environment.namespace(),
+        scaling_kind,
+        service.sanitized_name().as_str(),
+        replicas_count as u32,
```

Comment:
Even if not likely to happen, maybe worth being more cautious one this conversion?
What about using `usize::try_from(replicas_count).unwrap();`?